server.cpp

```diff
@@ -42,36 +47,117 @@ size_t read_complete(const char* buff, const error_code& err, size_t bytes)
+void send_string(tcp::socket& socket, const std::string& str)
 {
-    std::string error_message;
+    boost::asio::write(socket, boost::asio::buffer(str));
+}
 
-    if (request.verb() != "GET")
-        error_message = "HTTP/1.1 405 Method not allowed\r\n";
-    if (request.version() != "HTTP/1.1" && request.version() != "HTTP/1.0")
-        error_message += "HTTP/1.1 505 HTTP Version Not Supported\r\n";
+void send_index(tcp::socket& socket, DIR *dir, const std::string& path)
+{
+    std::string lines;
+
+    for (struct dirent *entry = readdir(dir); entry != NULL; entry = readdir(dir))
+    {
+        if (strcmp(".", entry->d_name) && strcmp("..", entry->d_name))
+        {
+            const std::string file_name = entry->d_name;
+
+            struct stat st;
+            if (stat((path + "/" + file_name).c_str(), &st) < 0)
+            {
+                lines = lines + "<tr><td>" + file_name + "</td><td>?</td><td>?</td></tr>";
+            }
+            else
+            {
+                const std::string file_date = ctime(&st.st_ctime);
+                lines = lines + "<tr><td><p><a href=\"" + file_name + "\">" + file_name + "</a></p></td><td>" + std::to_string(st.st_size) + "</td><td>" + file_date + "</td></tr>";
+            }
+        }
+    }
+
+    const std::string table_html ="<!DOCTYPE html> \
+        <html> \
+        <body> \
+        <table border=\"1\" cellspacing=\"0\" cellpadding=\"5\"> \
+        <tr><td>File name</td><td>File size</td><td>Last modification date</td></tr>" + lines + " \
+        </table> \
+        </body> \
+        </html>";
 
-    error_code ignored_error;
+    std::string index =  "HTTP/1.1 200 OK\r\nContent-Type: text/html; charset=utf-8\r\n\r\n" + table_html;
 
-    if (!error_message.empty())
+    send_string(socket, index);
+}
+
+void write_file(tcp::socket& socket, const std::string& request_path_file, const std::string& path)
+{
+    int fd = open((path + "/" + request_path_file).c_str(), O_RDONLY);
+
+    if (fd == -1)
     {
-        boost::asio::write(socket, boost::asio::buffer(error_message), ignored_error);
+        if (errno == ENOENT)
+            send_string(socket, "HTTP/1.1 404 File does not exist\r\nContent-Type: text/html\r\n\r\n404 File does not exist.\n");
+        else if (errno == EACCES)
+            send_string(socket, "HTTP/1.1 403 File access not allowed\r\nContent-Type: text/html\r\n\r\n403 File access not allowed.\n");
+        else
+            send_string(socket, "HTTP/1.1 500 File open error\r\nContent-Type: text/html\r\n\r\n500 File open error." + std::string(strerror(errno)) + "\n");
     }
     else
     {
-        const std::string message = "HTTP/1.1 200 OK\r\nHost: localhost\r\n\r\n" + date;
-        boost::asio::write(socket, boost::asio::buffer(message), ignored_error);
+        send_string(socket, "HTTP/1.1 200 OK\r\nContent-Type: text/html; charset=utf-8\r\nContent-Disposition: attachment\r\n\r\n");
```

Comment:
This is absolutely incorrect. You specify `Content-Type` as 'text/html' but instead you are sending files which may contain anything.